File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-mlx-4bit_think/20260830_231829/go/005_unit_test_writer/solution.go
package main

import (
	"fmt"
	"math"
)

func calculateDiscount(price float64, quantity int) float64 {
	if quantity < 10 {
		return price * float64(quantity)
	} else if quantity < 50 {
		return price * float64(quantity) * 0.9
	}
	return price * float64(quantity) * 0.75
}

func main() {
	const epsilon = 1e-9

	type testCase struct {
		description string
		price       float64
		quantity    int
		expected    float64
	}

	tests := []testCase{
		{
			description: "quantity 9 (below 10, no discount)",
			price:       100.0,
			quantity:    9,
			expected:    100.0 * 9, // 900.0
		},
		{
			description: "quantity 10 (boundary: exactly 10, 10% discount)",
			price:       100.0,
			quantity:    10,
			expected:    100.0 * 10 * 0.9, // 900.0
		},
		{
			description: "quantity 49 (below 50, 10% discount)",
			price:       100.0,
			quantity:    49,
			expected:    100.0 * 49 * 0.9, // 441.0
		},
		{
			description: "quantity 50 (boundary: exactly 50, 25% discount)",
			price:       100.0,
			quantity:    50,
			expected:    100.0 * 50 * 0.75, // 3750.0
		},
	}

	for _, tc := range tests {
		got := calculateDiscount(tc.price, tc.quantity)
		if math.Abs(got-tc.expected) < epsilon {
			fmt.Printf("PASS: %s\n", tc.description)
		} else {
			fmt.Printf("FAIL: %s (expected: %v, got: %v)\n", tc.description, tc.expected, got)
		}
	}
}

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-mlx-4bit_think/20260830_231829/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

interface TestCase {
    description: string;
    price: number;
    quantity: number;
    expected: number;
}

const testCases: TestCase[] = [
    {
        description: "quantity 9 (below discount threshold)",
        price: 100,
        quantity: 9,
        expected: 900, // 100 * 9
    },
    {
        description: "quantity 10 (first discount tier starts)",
        price: 100,
        quantity: 10,
        expected: 900, // 100 * 10 * 0.9
    },
    {
        description: "quantity 49 (just below second discount tier)",
        price: 100,
        quantity: 49,
        expected: 4410, // 100 * 49 * 0.9
    },
    {
        description: "quantity 50 (second discount tier starts)",
        price: 100,
        quantity: 50,
        expected: 3750, // 100 * 50 * 0.75
    },
];

let allPassed = true;

for (const tc of testCases) {
    const got = calculateDiscount(tc.price, tc.quantity);
    // Use a small epsilon for floating point comparison
    const isPass = Math.abs(got - tc.expected) < 1e-9;
    if (isPass) {
        console.log(`PASS: ${tc.description}`);
    } else {
        allPassed = false;
        console.log(`FAIL: ${tc.description} (expected: ${tc.expected}, got: ${got})`);
    }
}

if (allPassed) {
    console.log("\nAll tests passed.");
} else {
    console.log("\nSome tests failed.");
}